core/src/main/scala/org/apache/spark/io/CompressionCodec.scala

```diff
@@ -148,7 +148,7 @@ class SnappyCompressionCodec(conf: SparkConf) extends CompressionCodec {
   try {
     Snappy.getNativeLibraryVersion
   } catch {
-    case e: Error => throw new IllegalArgumentException
+    case e: Error => throw new IllegalArgumentException(e)
```

Comment:
Looks fine to me. I can see why just not catching the Error and letting it fly works as well. For now, fine to just preserve the same behavior but chain the cause.